### PR TITLE
OTC-63: Acquiring control numbers before enrolment

### PR DESCRIPTION
**Pull request by druchniewicz**
Related ticket: https://openimis.atlassian.net/browse/OTC-63

### CODE REVIEW

File: app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java [hunk 1]
```diff
@@ -2263,6 +2268,20 @@ public JSONArray getRecordedPoliciesWithIdentifier(String InternalIdentifier) {
         return RecordedPolicies;
     }
 
+    public JSONArray getNotEnrolledPolicies(String insuranceNumber, String insuranceProduct, String renewal) {
+        String aRenewal = "";
+        if (!renewal.equals("")) {
+            aRenewal = " AND isDone == '"+renewal+"'";
+        }
+
+        String uploaded = " AND UploadedDate == ''";
```
Comment (dragos-dobre) [inline app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java:2277]: Why you decomposed the SQL in substrings and then you concatenate even if you don't have conditions?

File: app/src/main/java/org/openimis/imispolicies/OverViewControlNumbers.java [hunk 2]
```diff
@@ -527,7 +527,7 @@ private int getControlNumber(final JSONArray order) throws IOException, JSONExce
         Thread thread = new Thread(){
             public void run() {
                 HttpClient httpClient = new DefaultHttpClient();
-                HttpPost httpPost = new HttpPost(AppInformation.DomainInfo.getDomain()+"/restapi/api/GetAssignedControlNumbers");
+                HttpPost httpPost = new HttpPost(AppInformation.DomainInfo.getDomain()+"/api/api/GetAssignedControlNumbers");
```
Comment (dragos-dobre) [inline app/src/main/java/org/openimis/imispolicies/OverViewControlNumbers.java:530]: Replace 'api' with 'restapi' as mentioned before.

File: app/src/main/java/org/openimis/imispolicies/OverViewPolicies1.java [hunk 3]
```diff
@@ -583,11 +565,10 @@ public void onClick(DialogInterface dialog,int id) {
         alertDialog.show();
     }
     private int getControlNumber(final JSONObject order, final String SmsRequired) throws IOException {
-
         Thread thread = new Thread(){
             public void run() {
                 HttpClient httpClient = new DefaultHttpClient();
-                HttpPost httpPost = new HttpPost(AppInformation.DomainInfo.getDomain()+"/restapi/api/GetControlNumber");
+                HttpPost httpPost = new HttpPost(AppInformation.DomainInfo.getDomain()+"/api/api/GetControlNumber");
```
Comment (dragos-dobre) [inline app/src/main/java/org/openimis/imispolicies/OverViewPolicies1.java:571]: Replace 'api' with 'restapi' as mentioned before.

File: app/src/main/java/org/openimis/imispolicies/ToRestApi.java [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 public class ToRestApi {
     Token tokenl = new Token();
-    private String uri = AppInformation.DomainInfo.getDomain()+"/restapi/";
+    private String uri = AppInformation.DomainInfo.getDomain()+"/api/";
```
Comment (dragos-dobre) [inline app/src/main/java/org/openimis/imispolicies/ToRestApi.java:25]: Replace 'api' with 'restapi'.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -380,4 +380,8 @@
     <string name="date_to">Date To</string>
     <string name="send_sms">Send SMS</string>
     <string name="requested_sms">Requested SMS</string>
+    <string name="enrolled_policies">Enrolled Policies</string>
```
Comment (dragos-dobre) [inline app/src/main/res/values/strings.xml:383]: Did you added these keys in Lokalize?

Comment (druchniewicz) [reply app/src/main/res/values/strings.xml:383]: You mean that these keys should be added in Translations Editor which will automatically add entries in "values-fr/strings.xml" file ?

File: app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java (hunk 1 above)
Comment (druchniewicz) [reply app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java:2277]: Yes, you're right. Without any conditions these additional variables are not needed and can be included directly in query string

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (druchniewicz) [reply app/src/main/res/values/strings.xml:383]: No I didn't. Damian give me a link to policies app in Lokalize - https://app.lokalise.com/project/260745865d22f770125fe4.66491371/ but I don't have a permissions. Could you give me an access to this resosurces? druchniewicz@soldevelo.com

Comment (dragos-dobre) [reply app/src/main/res/values/strings.xml:383]: You have now access to Lokalize.

Comment (dragos-dobre) [reply app/src/main/res/values/strings.xml:383]: Don't forget to add the keys in French too.

Comment (dragos-dobre) [reply app/src/main/res/values/strings.xml:383]: I update the French translation on Lokalize to be synchronised.

**Pull request merged**